AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift

```diff
@@ -38,11 +38,18 @@ class SDCardMobileSessionsSavingService: SDCardMobileSessionssSaver {
     @Injected private var persistenceController: PersistenceController
     private var databaseStorage = SDSyncMobileSessionsDatabaseStorage()
     private lazy var context: NSManagedObjectContext = persistenceController.editContext
+    var sessionUUID: SessionUUID? = .init()
+    private var location: CLLocationCoordinate2D? = nil
+    private var deviceLocation: CLLocationCoordinate2D? = nil
     
-    func saveDataToDb(filesDirectoryURL: URL, deviceID: String, completion: @escaping (Result<Void, Error>) -> Void) {
+    func saveDataToDb(filesDirectoryURL: URL, deviceID: String, deviceLocation: CLLocationCoordinate2D, completion: @escaping (Result<Void, Error>) -> Void) {
         var isDirectory: ObjCBool = false
         guard FileManager.default.fileExists(atPath: filesDirectoryURL.path, isDirectory: &isDirectory) else { completion(.success(())); return }
         
+        if self.deviceLocation == nil {
+            self.deviceLocation = deviceLocation
+        }
```

Comment:
you can make it a one-liner like this probably: 
`self.deviceLocation = self.devicelLocation ?? deviceLocation`